src/decorations/functions.jl

```diff
@@ -0,0 +1,315 @@
+            (isthin(q) && isinteger(q.lo) && q.lo > zero(q)) ||
+            (isthin(q) && isinteger(q.lo) && zero(T) ∉ x)
+        return DecoratedInterval(r, d)
+    end
+    DecoratedInterval(r, trv)
+end
+
+## Discontinuous functions (sign, ceil, floor, trunc) and round
+function sign{T}(xx::DecoratedInterval{T})
+    r = sign(interval(xx))
+    d = decoration(xx)
+    isthin(r) && return DecoratedInterval(r, d)
+    DecoratedInterval(r, decay(d,def))
+end
+function ceil{T}(xx::DecoratedInterval{T})
+    x = interval(xx)
+    r = ceil(x)
+    d = decoration(xx)
+    if isinteger(x.hi)
+        d = decay(d, dac)
+    end
+    isthin(r) && return DecoratedInterval(r, d)
+    DecoratedInterval(r, decay(d,def))
+end
+function floor{T}(xx::DecoratedInterval{T})
+    x = interval(xx)
+    r = floor(x)
+    d = decoration(xx)
+    if isinteger(x.lo)
+        d = decay(d, dac)
+    end
+    isthin(r) && return DecoratedInterval(r, d)
+    DecoratedInterval(r, decay(d,def))
+end
+function trunc{T}(xx::DecoratedInterval{T})
+    x = interval(xx)
+    r = trunc(x)
+    d = decoration(xx)
+    if (isinteger(x.lo) && x.lo < zero(T)) || (isinteger(x.hi) && x.hi > zero(T))
+        d = decay(d, dac)
+    end
+    isthin(r) && return DecoratedInterval(r, d)
+    DecoratedInterval(r, decay(d,def))
+end
+
+function round(xx::DecoratedInterval, ::RoundingMode{:Nearest})
+    x = interval(xx)
+    r = round(x)
+    d = decoration(xx)
+    if isinteger(2*x.lo) || isinteger(2*x.hi)
+        d = decay(d, dac)
+    end
+    isthin(r) && return DecoratedInterval(r, d)
+    DecoratedInterval(r, decay(d,def))
+end
+function round(xx::DecoratedInterval, ::RoundingMode{:NearestTiesAway})
+    x = interval(xx)
+    r = round(x,RoundNearestTiesAway)
+    d = decoration(xx)
+    if isinteger(2*x.lo) || isinteger(2*x.hi)
+        d = decay(d, dac)
+    end
+    isthin(r) && return DecoratedInterval(r, d)
+    DecoratedInterval(r, decay(d,def))
+end
+round(xx::DecoratedInterval) = round(xx, RoundNearest)
+round(xx::DecoratedInterval, ::RoundingMode{:ToZero}) = trunc(xx)
+round(xx::DecoratedInterval, ::RoundingMode{:Up}) = ceil(xx)
+round(xx::DecoratedInterval, ::RoundingMode{:Down}) = floor(xx)
+
+
+## Define binary functions with no domain restrictions
+binary_functions = ( :min, :max )
+
+for f in binary_functions
+    @eval function $(f){T}(xx::DecoratedInterval{T}, yy::DecoratedInterval{T})
+        r = $f(interval(xx), interval(yy))
+        d = decay(decoration(r), decoration(xx), decoration(yy))
+        DecoratedInterval(r, d)
+    end
+end
+
+## abs
+abs{T}(xx::DecoratedInterval{T}) =
+    DecoratedInterval(abs(interval(xx)), decoration(xx))
+
+
+## Other (cancel and set) functions
+other_functions = ( :cancelplus, :cancelminus, :intersect, :hull, :union )
+
+for f in other_functions
+    @eval $(f){T}(xx::DecoratedInterval{T}, yy::DecoratedInterval{T}) =
+        DecoratedInterval($(f)(interval(xx), interval(yy)), trv)
```

Comment:
It is discussed in the standard; there, it is recommended that the user adapts it by changing the decoration. I'll add this (and other) docstrings explaining this.